src/visualizers/VisualizerFactorHistogram.ts

```diff
@@ -0,0 +1,200 @@
+            value: this.firstIndex,
+            forceType: 'integer',
+            displayName: 'First Index',
+            required: true,
+        },
+
+        terms: {
+            value: this.terms,
+            forceType: 'integer',
+            displayName: 'How many terms of the series',
+            required: true,
+        },
+    }
+
+    checkParameters() {
+        const status = super.checkParameters()
+
+        if (this.params.binSize.value < 1) {
+            status.isValid = false
+            status.errors.push('Bin Size can not be less than 1')
+        }
+
+        return status
+    }
+
+    // Obtain the true first index
+    startIndex(): number {
+        return Math.max(this.firstIndex, this.seq.first)
+    }
+
+    // Obtain the true number of terms
+    numTerms(): number {
+        return Math.min(this.terms + this.startIndex(), this.seq.last)
+    }
+
+    // Find the largest value of the sequence asked for
+    largestValue(): bigint {
+        let largest_value = this.seq.getElement(this.startIndex())
+        for (let i = this.startIndex(); i < this.numTerms(); i++) {
+            const value = this.seq.getElement(i)
+            if (value > largest_value) {
+                largest_value = value
+            }
+        }
+        return largest_value
+    }
+
+    // Create an array with the number of factor of
+    // each element at the corresponding index of the array
+    factorArray(): number[] {
+        const factorArray = []
+        for (let i = this.startIndex(); i < this.numTerms(); i++) {
+            let counter = 0
+            const factors = this.seq.getFactors(i)
+            if (factors) {
+                for (const [base, power] of factors) {
+                    if (base > 0) {
+                        counter += Number(power)
+                    } else if (base === 0n) {
+                        counter = 0
+                    }
+                }
+            }
+
+            factorArray[i] = counter
+        }
+        return factorArray
+    }
+
+    // Create an array with the frequency of each number
+    // of factors in the corresponding bins
+    binFactorArray(): number[] {
+        const binFactorArray = []
+        const factorArray = this.factorArray()
+        for (
+            let i = 0;
+            i < natlog(this.largestValue()) / natlog(2) + 1;
+            i++
+        ) {
+            binFactorArray.push(0)
+        }
+
+        for (let i = 0; i < factorArray.length; i++) {
+            binFactorArray[Math.trunc(factorArray[i] / this.binSize)]++
+        }
+
+        return binFactorArray
+    }
+
+    // Create a number that represents how
+    // many pixels wide each bin should be
+    binWidth(): number {
+        const binFactorArray = this.binFactorArray()
+        let array_length = 0
+        for (let i = 0; i < binFactorArray.length; i++) {
+            if (binFactorArray[i] != 0) {
+                array_length = i
+            }
+        }
```

Comment:
Once you have switched to computing the largest element of factorArray, you will be able to compute the number of bins directly, no loop needed.